src/toil/test/wdl/builtinTest.py

```diff
@@ -321,7 +323,13 @@ def test_write(self):
         self.check_function('write_map', cases=['as_command'],
                             expected_result='key1\tvalue1\nkey2\tvalue2')
 
+    def test_range(self):
+        # NOTE: this test depends on write_lines().
+        self.check_function('range', cases=['as_input'], expected_result='0\n1\n2\n3\n4\n5\n6\n7')
```

Comment:
Consider adding a test for `0` and `-1` as well.

For -1, `with self.assertRaises(AssertionError):` is useful.  `AssertionError` is fine, though if you want, it might be worth making it a `RuntimeError` or a WDL specific error class.

Reply:
I added `WDLRuntimeError` for WDL-related runtime errors. I should probably replace the other asserts at some point. 

Also, it turns out `with self.assertRaises(...)` doesn't work. We're checking the output via subprocess, so `subprocess.CalledProcessError` is thrown when the command fails. Although, the subprocess exception seems to be consumed by the unittest `TestCase` so we end up with nothing to catch. 

I'll see if there's anything else I can do to add the test for `-1`.